src/resources/2024/ChestMimic.ts

```diff
@@ -130,3 +132,18 @@ export function differentiate(
   runCombat(...combat);
   return true;
 }
+
+/**
+ * Check how many of a monster is available to differentiate into
+ *
+ * @param monster The monster to differentiate your egg into; may behave weirdly with name collisions
+ * @returns How many of a Monster we can differentiate
+ */
+export function differentiableQuantity(monster: Monster): number {
+  if (!have_($item`mimic egg`)) return 0;
+  const regex = new RegExp(`${monster.name}\\s*(?:\\((\\d+)\\))?`, "i");
```

Comment:
I would not make this regex case-insensitive, it makes this more prone to errors

Reply:
also, I have an extremely funny test case: do you know if it uses commas when it hits 4 digits?

Reply:
I have no idea! It would take 100 days to find out